pkg/kube/controllers/extendedjob/job_create_secret.go

```diff
@@ -0,0 +1,286 @@
+
+	for {
+		time.Sleep(5 * time.Second)
+		_, err := os.Stat(filePath)
+		if err != nil {
+			if os.IsNotExist(err) {
+				continue
+			} else {
+				errorChannel <- err
+				break
+			}
+		}
+		fileNotifyChannel <- containerIndex
+		break
+	}
+}
+
+// authenticateInCluster authenticates with the in cluster and returns the client
+func authenticateInCluster() (*kubernetes.Clientset, *versioned.Clientset, error) {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "failed to authenticate with incluster config")
+	}
+
+	clientSet, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "failed to create clientset with incluster config")
+	}
+
+	versionedClientSet, err := versioned.NewForConfig(config)
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "failed to create versioned clientset with incluster config")
+	}
+
+	return clientSet, versionedClientSet, nil
+}
+
+func createOutputSecret(outputContainer corev1.Container, namePrefix string, namespace string, podName string, clientSet *kubernetes.Clientset, exjob *ejv1.ExtendedJob) error {
+
+	secretName := namePrefix + outputContainer.Name
+
+	// Fetch json from file
+	filePath := fmt.Sprintf("%s%s%s", "/mnt/quarks/", outputContainer.Name, "/output.json")
+	file, _ := ioutil.ReadFile(filePath)
+	var data map[string]string
+	err := json.Unmarshal([]byte(file), &data)
+	if err != nil {
+		return errors.Wrapf(err, "failed to convert output file %s into json for creating secret %s in pod %s",
+			filePath, secretName, podName)
+	}
+
+	// Create secret for the outputfile to persist
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: namespace,
+		},
+	}
+	secretLabels := exjob.Spec.Output.SecretLabels
+	if secretLabels == nil {
+		secretLabels = map[string]string{}
+	}
+	secretLabels[ejv1.LabelPersistentSecretContainer] = outputContainer.Name
+	if ig, ok := podutil.LookupEnv(outputContainer.Env, converter.EnvInstanceGroupName); ok {
+		secretLabels[ejv1.LabelInstanceGroup] = ig
+	}
+
+	if exjob.Spec.Output.Versioned {
+		// Use secretName as versioned secret name prefix: <secretName>-v<version>
+		err = createVersionSecret(clientSet, namespace, exjob.GetName(), exjob.GetUID(), secretName, data, secretLabels, "created by extendedjob")
+		if err != nil {
+			return errors.Wrapf(err, "could not persist ejob's %s output to a secret", exjob.GetName())
+		}
+	} else {
+		secret.StringData = data
+		secret.Labels = secretLabels
+
+		_, err = clientSet.CoreV1().Secrets(namespace).Create(secret)
+
+		if err != nil {
+			if apierrors.IsAlreadyExists(err) {
+				// If it exists update it
+				_, err = clientSet.CoreV1().Secrets(namespace).Update(secret)
+				if err != nil {
+					return errors.Wrapf(err, "Failed to update secret %s for container %s in pod %s.", secretName, outputContainer.Name, podName)
```

Comment:
shouldn't capitalize first letter in an error

Reply:
Done.